playbooks/robusta_playbooks/k8s_resource_enrichments.py

```diff
@@ -43,32 +46,103 @@ def to_pod_row(pod: Pod, cluster_name: str) -> List:
+    kind: str = resource.kind or ""
+    if kind not in supported_resources:
         raise ActionException(
-            ErrorCodes.RESOURCE_NOT_SUPPORTED, f"Related pods is not supported for resource {resource.kind}"
+            ErrorCodes.RESOURCE_NOT_SUPPORTED, f"Related pods is not supported for resource {kind}"
         )
 
-    if resource.kind == "Job":
+    if kind == "Job":
         job_pods = get_job_all_pods(resource)
         pods = job_pods if job_pods else []
-    elif resource.kind == "Pod":
+    elif kind == "Pod":
         pods = [resource]
-    elif resource.kind == "Node":
+    elif kind == "Node":
         pods = Pod.listPodForAllNamespaces(field_selector=f"spec.nodeName={resource.metadata.name}").obj.items
     else:
         selector = build_selector_query(resource.spec.selector)
         pods = PodList.listNamespacedPod(namespace=resource.metadata.namespace, label_selector=selector).obj.items
 
+    return pods
+
+def to_pod_obj(pod: Pod, cluster: str) -> Dict[str,Any]:
+    resource_requests = pod_requests(pod)
+    resource_limits = pod_limits(pod)
+    addresses = ",".join([address.ip for address in pod.status.podIPs])
+    return {
+        "name": pod.metadata.name,
+        "namespace": pod.metadata.namespace,
+        "node": pod.spec.nodeName,
+        "clusterName": cluster,
+        "cpuLimit": resource_limits.cpu,
+        "cpuRequest": resource_requests.cpu,
+        "memoryLimit": resource_limits.memory,
+        "memoryRequest": resource_requests.memory,
+        "creationTime": pod.metadata.creationTimestamp,
+        "restarts": pod_restarts(pod),
+        "addresses": addresses,
+        "containers": get_pod_containers(pod),
+        "status": pod.status.phase,
+    }
+
+def get_pod_containers(pod: Pod) -> List[Dict[str,Any]]:
+    containers = []
+    for container in pod.spec.containers:
+        requests = PodContainer.get_requests(container)
+        limits = PodContainer.get_limits(container)
+        containerStatus: Optional[ContainerStatus] = PodContainer.get_status(pod, container.name)  
+        currentState : Optional[ContainerState] = getattr(containerStatus, "state", None)
+        stateStr : str = "waiting"
+        state = None
+        if currentState:
+            for s in ["running", "waiting" , "terminated"]:
+                state = getattr(currentState, s, None)
+                if state is not None:
+                    stateStr = s
+                    break
+
+        containers.append({
+            "name": container.name,
+            "cpuLimit": limits.cpu,
+            "cpuRequest": requests.cpu,
+            "memoryLimit": limits.memory,
+            "memoryRequest": requests.memory,
+            "restarts": getattr(containerStatus, "restartCount", 0),
+            "status": stateStr,
+            "created": getattr(state, "startedAt", None)
+        })
+    
+    return containers
+ 
+
+
+
+@action
+def resource_related_pods(event: KubernetesResourceEvent):
```

Comment:
is the only difference that you added containers?
It might be word it to add the whole yaml so we dont have to make a new related pods for every new field we want.